### DISCUSSION TITLE
Issue installing Machinaris on Linux

### DISCUSSION THREAD

**Discussion by olivierLA75** [guydavis/machinaris#181] [Q&A] [answered]
Hi, so installed Docker and ran the hello world without problem but when i try the following commands it doesn't work, any idea of what's the issue?

chia1@CHIA1-Precision-Tower-7910:~$ docker run --name='machinaris' -d -t -h $(hostname -s) \
>     -p 8926:8926 -p 8927:8927 -p 8444:8444 -p 8447:8447 -p 6888:6888 -p 6885:6885 \
>     -e TZ="America/Los_Angeles" \
>     -v '~/.machinaris':'/root/.chia':'rw' \
>     -v '/dev/sdb2':'/plots':'rw' \
>     -v '/dev/nvme0n1p3':'/plotting':'rw' \ 
docker: invalid reference format.
See 'docker run --help'.
chia1@CHIA1-Precision-Tower-7910:~$     ghcr.io/guydavis/machinaris
bash: ghcr.io/guydavis/machinaris: No such file or directory
chia1@CHIA1-Precision-Tower-7910:~$

**Comment by guydavis** [comment, marked as answer]
Good day!  Based on what you pasted it looks like you left off the image name (end of the command) so the Docker run command failed.  Instead the image name was entered as the next bash command which the shell couldn't find.

Try to paste the entire Docker run into your shell at once.  Finally, I wouldn't recommend treating the `/dev/*` paths as mounted drives.  Please review [Mounting Disks](https://help.ubuntu.com/community/AutomaticallyMountPartitions) on Ubuntu.

Example: 

![image](https://user-images.githubusercontent.com/12936/125848528-3f6029a7-c4b5-4e68-96f8-c41df9310eb4.png)

If this answers your question, please mark the Discussion as Answered.  Thanks!

**Comment by olivierLA75** [comment]
Thanks for your reply. I keep having the same message saying file ‘docker’ in directory has no filename extension (see attached)
Do you know what it could be?

**Reply by guydavis** [reply]
Sorry, I don't see an attachment.  Are you still having trouble?

**Comment by sadreddini** [comment]
Hi,
Actually I had the same problem but with answer from @guydavis to the first question I managed to move forward one step but now I get below error message, your suggestions are appreciated.
PS: I am totally new to Linux and Docker stuff.

![image](https://user-images.githubusercontent.com/87619085/126081271-84685e44-f774-4b8e-b2a2-9242a8ff9938.png)

**Reply by guydavis** [reply]
Good day! This is expected if you run Machinaris twice with only the first process being allowed to bind tcp/8444.  On second launch, (first still running in background), the attempt to bind ports will fail with error above.  To re-run Machinaris try:

```
docker stop machinaris
docker rm machinaris
docker run ...
```

Hope this helps!

**Comment by sadreddini** [comment]
Hi,
Very good day (or perhaps night in there) to you.
First of all I would like to express my appreciation to your efforts and time spending to develop Machinaris and also replying the questions.
I got it up and running now with your help although I did reset my pc before posting the first comment and had the same problem.
I still didn't play with it but from few videos and articles I assume it is a very professional and comprehensive Chia plotting/farming/harvesting manager which is not paid attention to enough yet by users.
Thanks a lot.
